packages/contracts-bedrock/src/L1/SystemConfigInterop.sol

```diff
@@ -8,12 +8,31 @@ import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 import { SystemConfig } from "src/L1/SystemConfig.sol";
 import { ConfigType } from "src/L2/L1BlockInterop.sol";
 import { StaticConfig } from "src/libraries/StaticConfig.sol";
+import { Storage } from "src/libraries/Storage.sol";
 
 /// @title SystemConfigInterop
 /// @notice The SystemConfig contract is used to manage configuration of an Optimism network.
 ///         All configuration is stored on L1 and picked up by L2 as part of the derviation of
 ///         the L2 chain.
 contract SystemConfigInterop is SystemConfig {
+    /// @notice Storage slot where the address authorized to call `addDependency` is stored
+    bytes32 internal constant ADD_DEPENDENCY_ROLE_HOLDER_SLOT =
+        bytes32(uint256(keccak256("systemconfig.adddependencyrole")) - 1);
+
+    /// @notice Storage slot where the address authorized to call `removeDependency` is stored
+    bytes32 internal constant REMOVE_DEPENDENCY_ROLE_HOLDER_SLOT =
+        bytes32(uint256(keccak256("systemconfig.removedependencyrole")) - 1);
+
+    /// @notice Storage slot where the foundation multisig address is stored
+    bytes32 internal constant FOUNDATION_MULTISIG_SLOT =
+        bytes32(uint256(keccak256("systemconfig.foundationmultisig")) - 1);
+
+    /// @notice Initializer.
+    /// @param _foundationMultisig The Foundation's multisig address
+    function initialize(address _foundationMultisig) external initializer {
```

Comment:
We will want to wrap a call to this `initialize` so we will need all the same args + the `dependencyManager`

https://github.com/ethereum-optimism/optimism/blob/bf4693e76db85148121aea35904755672a153449/packages/contracts-bedrock/src/L1/SystemConfig.sol#L179

Reply:
This would basically be:

```solidity
struct Roles {
  address chainOperator;  // used to be owner
  address dependencyManager;
  address batcherAddress;
  address unsafeBlockSigner;
}

    function initialize(
        uint32 _basefeeScalar,
        uint32 _blobbasefeeScalar,
        uint64 _gasLimit,
        Roles memory _roles,
        ResourceMetering.ResourceConfig memory _config,
        address _batchInbox,
        SystemConfig.Addresses memory _addresses
    ) {
  // call other initializer
  initializer(...);
  // initialize new argument dependencyManager
}
```

Reply:
The reason why we are doing it this way is that it is an experimental feature and we don't want to add the code directly into the mainline contracts yet. Once interop is more finalized, we will merge the interop contracts into the mainline contracts

https://github.com/ethereum-optimism/design-docs/blob/6e55f59b7548cbd6abd7d84c9e0e50fca6a48421/smart-contract-feature-development.md

Reply:
[Here in the smart-contract-feature-development readme](https://github.com/ethereum-optimism/design-docs/blob/6e55f59b7548cbd6abd7d84c9e0e50fca6a48421/smart-contract-feature-development.md#handling-inheritance), it says that virtual keyword should be added to functions that need to be modified. what if modifying involves changing/reordering the arguments (basically anything that means that the function sig changes and virtual keyword does not solve it, similar to the initialize function now), how is this handled? a virtual modifier to the old function and reimplementing it to revert when called? Or is this just ignored until its time to merge the old and new-feature version then it is removed?

Reply:
We spoke about this async, but in this case we want to wrap the new `initialize` call around the old `initialize` call